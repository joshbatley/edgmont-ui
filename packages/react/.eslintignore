**/lib/*
**/node_modules/
**/jest.config.js
**/setupTests.ts
**/rollup.config.js
**/dist/*
**/src/tailwind.css
**/docs/*
**/preset.js
