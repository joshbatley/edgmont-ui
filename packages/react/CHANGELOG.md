# @edgmont-ui/react

## 1.0.0

### Major Changes

- b4b59a8: Revamped all components, add themes and fix broken logix

## 0.3.0

### Minor Changes

- cc0ffd7: Updated readme and version

## 3.0.1

### Patch Changes

- 1c12ec6: readme

## 3.0.0

### Major Changes

- 52e9e42: asdas

## 2.0.2

### Patch Changes

- 7e13337: asd

## 2.0.1

### Patch Changes

- cba70fc: asdada

## 2.0.0

### Major Changes

- 323e843: ad

### Minor Changes

- a80937b: asd

## 1.0.0

### Major Changes

- c4b8b3f: asd
