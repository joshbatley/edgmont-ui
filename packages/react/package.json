{
  "name": "@edgmont-ui/react",
  "version": "1.0.2",
  "private": false,
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "homepage": "https://ui.edgmont.dev/",
  "bugs": {
    "url": "https://github.com/joshbatley/edgmont-ui/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/joshbatley/edgmont-ui"
  },
  "jest": {
    "preset": "@repo/jest-presets/jest/browser"
  },
  "contributors": [
    "joshbatley"
  ],
  "files": [
    "dist",
    "package.json"
  ],
  "scripts": {
    "build": "tsup src/index.ts --format esm,cjs --dts --external react",
    "dev": "tsup src/index.ts --format esm,cjs --watch --dts --external react",
    "lint": "eslint . --max-warnings 0",
    "test": "jest",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@repo/eslint-config": "workspace:*",
    "@repo/jest-presets": "workspace:*",
    "@repo/typescript-config": "workspace:*",
    "@turbo/gen": "^1.13.3",
    "@types/eslint": "^8.56.10",
    "@types/node": "^20.12.12",
    "@types/react": "^18.3.2",
    "@types/react-dom": "^18.3.0",
    "eslint": "^8.57.0",
    "react": "^18.3.1",
    "typescript": "5.4.5"
  },
  "dependencies": {
    "@heroicons/react": "^2.1.3",
    "@popperjs/core": "^2.11.8",
    "@testing-library/jest-dom": "^6.4.5",
    "@testing-library/react": "^14.3.1",
    "@testing-library/react-hooks": "^8.0.1",
    "@testing-library/user-event": "^14.5.2",
    "@types/jest": "^29.5.12",
    "@types/react-datepicker": "^6.2.0",
    "@types/styled-components": "^5.1.34",
    "@types/styled-system": "^5.1.22",
    "date-fns": "^3.6.0",
    "downshift": "^8.5.0",
    "jest": "^29.7.0",
    "jest-styled-components": "^7.2.0",
    "polished": "^4.3.1",
    "prism-react-renderer": "^2.3.1",
    "react-datepicker": "^6.9.0",
    "react-dom": "^18.3.1",
    "react-dropzone": "^14.2.3",
    "react-focus-lock": "^2.12.1",
    "react-popper": "^2.3.0",
    "react-simple-code-editor": "^0.13.1",
    "react-spring": "^9.7.3",
    "react-use": "^17.5.0",
    "styled-components": "^5.3.11",
    "styled-system": "^5.1.5",
    "tsup": "^8.0.2"
  }
}