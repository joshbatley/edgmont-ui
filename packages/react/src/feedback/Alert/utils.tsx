import React from 'react';
import {
  ExclamationCircleIcon,
  ExclamationTriangleIcon,
  InformationCircleIcon,
  CheckCircleIcon,
} from '@heroicons/react/24/solid';

export const getIcon = (severity: Severity) => {
  let props = {
    height: 22,
    width: 22,
  };
  switch (severity) {
    case 'error':
      return <ExclamationCircleIcon {...props} />;
    case 'warning':
      return <ExclamationTriangleIcon {...props} />;
    case 'info':
      return <InformationCircleIcon {...props} />;
    case 'success':
      return <CheckCircleIcon {...props} />;
  }
};

