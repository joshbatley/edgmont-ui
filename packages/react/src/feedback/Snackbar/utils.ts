import { IntercardinalPoints } from '../../types';

export const createId = () => {
  return Math.random().toString(36).substr(2, 9);
};

export const getPosition = (position: IntercardinalPoints) => {
  switch (position) {
    case 'NE':
      return `
      top: 0;
      right: 0;
      flex-direction: column;
      `;
    case 'NW':
      return `
      top: 0;
      left: 0;
      flex-direction: column;
      `;
    case 'SE':
      return `
      right: 0;
      bottom: 0;
      flex-direction: column-reverse;
      `;
    case 'SW':
      return `
        left: 0;
        bottom: 0;
        flex-direction: column-reverse;
      `;
  }
};
