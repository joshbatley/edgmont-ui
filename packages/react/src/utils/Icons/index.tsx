
export const RadioIcon = ({ ...rest }) => (
  <svg viewBox='0 0 16 16' fill='currentColor' xmlns='http://www.w3.org/2000/svg' {...rest}>
    <circle cx='8' cy='8' r='3' />
  </svg>
);
