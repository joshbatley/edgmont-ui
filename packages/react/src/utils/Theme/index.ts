export * from './Theme';
export * from './DefaultThemeProvider';
export * from './GlobalStyles';
