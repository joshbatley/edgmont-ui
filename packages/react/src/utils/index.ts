export * from './Backdrop';
export * from './ClickOutside';
export * from './FocusLock';
export * from './Portal';
export * from './Scrollable';
export * from './Transitions';
export * from './Icons';
export * from './Theme';
