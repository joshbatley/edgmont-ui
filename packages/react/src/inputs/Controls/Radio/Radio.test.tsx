import React from 'react';
import { render as testRender } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeDefaultProvider } from 'utils';
import { Radio } from '.';

function render(Comp: any) {
  return testRender(<ThemeDefaultProvider>{Comp}</ThemeDefaultProvider>);
}

const onChangeMock = jest.fn();

describe('Radio test', () => {
  it('should render', () => {
    const { container } = render(<Radio />);
    expect(container).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
  it('should render with disabled classes', () => {
    const { container } = render(<Radio disabled />);
    expect(container).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
  it('should errored with disabled classes', () => {
    const { container } = render(<Radio error />);
    expect(container).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });

  it('should update value when updated', async () => {
    const { getByRole } = render(<Radio value="testing-value" name="testing-input" onChange={onChangeMock} />);
    expect(onChangeMock).not.toHaveBeenCalled();
    await userEvent.click(getByRole('radio', { hidden: true }));
    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });
});
