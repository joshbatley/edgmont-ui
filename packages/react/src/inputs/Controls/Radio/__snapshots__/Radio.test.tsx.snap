// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio test should render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  color: foreground;
}

.c1 > :not([hidden]) ~ :not([hidden]) {
  margin-left: 0.5rem;
}

.c2 {
  position: relative;
  width: 1rem;
  height: 1rem;
  border: 1px solid hsl(0,0%,100%);
  background: hsl(0,0%,100%);
  overflow: hidden;
  border: 1px solid hsl(240,5.9%,10%);
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1),0 1px 2px -1px rgb(0 0 0 / 0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.5rem;
  border-radius: 9999px;
  opacity: 1;
}

.c5 {
  background: hsl(0,0%,100%);
  width: 100%;
  height: 100%;
  position: absolute;
}

input:checked ~ .c4 {
  background: hsl(240,5.9%,10%);
}

.c0:hover input:checked ~ .c4 {
  background: hsl(240,5.9%,10%);
}

.c7 svg {
  color: transparent;
  z-index: 10;
  position: relative;
}

.c0 input:checked ~ .c6 svg {
  color: hsl(0,0%,100%);
}

.c0:hover .c6 svg {
  color: hsl(240,5.9%,10%);
}

.c3 {
  display: none;
}

<div>
  <label
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <input
        class="c3"
        type="radio"
        value=""
      />
      <div
        class="c4 c5"
      />
      <div
        class="c6 c7"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="8"
            r="3"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Radio test should render with disabled classes 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
  color: hsl(240,3.8%,46.1%);
}

.c1 > :not([hidden]) ~ :not([hidden]) {
  margin-left: 0.5rem;
}

.c2 {
  position: relative;
  width: 1rem;
  height: 1rem;
  border: 1px solid hsl(0,0%,100%);
  background: hsl(0,0%,100%);
  overflow: hidden;
  border: 1px solid hsl(240,5.9%,10%);
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1),0 1px 2px -1px rgb(0 0 0 / 0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.5rem;
  border-radius: 9999px;
  opacity: 0.5;
}

.c5 {
  background: hsl(0,0%,100%);
  width: 100%;
  height: 100%;
  position: absolute;
}

input:checked ~ .c4 {
  background: hsl(240,5.9%,10%);
}

.c0:hover input:checked ~ .c4 {
  background: hsl(240,5.9%,10%);
}

.c7 svg {
  color: transparent;
  z-index: 10;
  position: relative;
}

.c0 input:checked ~ .c6 svg {
  color: hsl(0,0%,100%);
}

.c3 {
  display: none;
}

<div>
  <label
    class="c0 c1"
    disabled=""
  >
    <div
      class="c2"
      disabled=""
    >
      <input
        class="c3"
        disabled=""
        type="radio"
        value=""
      />
      <div
        class="c4 c5"
      />
      <div
        class="c6 c7"
        disabled=""
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="8"
            r="3"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;
